crochet/_shutdown.py

```diff
@@ -53,8 +54,15 @@ def run(self):
                 log.err()
 
 
-# This is... fragile. Not sure how else to do it though.
-_registry = FunctionRegistry()
-_watchdog = Watchdog([t for t in threading.enumerate()
-                     if t.name == "MainThread"][0], _registry.run)
-register = _registry.register
+def default_shutdown_condition():
```

Comment:
In retrospect it's not clear to me why this is module-level state at all. Seems like all of the state could live inside `EventLoop`, and then there's no need to pass a `FunctionRegistry` instance to `EventLoop.__init__`, it would just create one.